src/libostree/ostree-sysroot.c

```diff
@@ -259,6 +259,63 @@ ostree_sysroot_set_mount_namespace_in_use (OstreeSysroot  *self)
   self->mount_namespace_in_use = TRUE;
 }
 
+/**
+ * ostree_sysroot_initialize_mount_namespace:
+ *
+ * Prepare the current process for modifying a booted sysroot, if applicable.
+ * This function subsumes the functionality of `ostree_sysroot_initialize`
+ * and may be invoked wherever that function is.
+ *
+ * If the sysroot does not appear to be booted, or where the current process is not uid 0,
+ * this function returns successfully.
+ *
+ * Otherwise, if the process is in the same mount namespace as pid 1, create
+ * a new namespace.
+ *
+ * If you invoke this function, it must be before ostree_sysroot_load(); it may
+ * be invoked before or after ostree_sysroot_initialize().
+ *
+ * Since: 2022.7
+ */
+gboolean
+ostree_sysroot_initialize_mount_namespace (OstreeSysroot *self, GError **error)
```

Comment:
A more explicit `initialize_with_mount_namespace` maybe?

As we are adding a new API, do you want to add a `GCancellable` anyway? The `glnx` primitives would gladly take one (even if not really using it).

Reply:
Done and done.